### DISCUSSION TITLE
Github gets annoying: it must get better ! Too much not working projects. Github needs a review system.

### DISCUSSION THREAD

**Discussion by NeverBeLazyG** [community/community#141148] [Other Feature Feedback, Questions, & Ideas]
I just want to experience once that a developer actually tests their releases before publishing them. This endless tinkering with every Git project is so annoying.

lately (especially AI python based project), i tested about 20 Ai tools, and at lease 15 did not work out of the box, when following the instructions. and pc environment is a very rich environment, with a lot already installed librarys, tools, developer tools. i dont want to know how much problems people have who dont even have rich environments like me.

You follow the instructions.

And then 90% of the time there's some error that you end up spending hours fixing. the Guys, who releasing project should test their tools/programs/scripts in an fresh environment before releasing it.

I dont understand why people releasing projects which dont work. its just frustrating.

I really miss the good old days when you could just click on setup.exe, and it worked.

Github should find some solution for this problem. github gets more and more to a frustrating place.

at least make a 5-star rating system: where ppl can rate if the project installation, workflow etc. is worth it.

**Comment by Dhreetiman** [comment]
What is the actual purpose of it?

**Reply by NeverBeLazyG** [reply]
to increase the quality of releases. if there is a rating system the quality of releases could be increased.

**Comment by longnh238** [comment]
I would agree. Many developers don't test their releases in clean environments, leading to dependency issues and errors. The solution is for developers to test in fresh environments using tools like Docker or virtual environments and automate testing with CI/CD pipelines. A project rating system with filtering, as you mentioned, could also help users find reliable repos faster.

**Reply by NeverBeLazyG** [reply]
yes that what i think. github really needs that. there are so much very bad trash projects. which lets github not look good. there should be a rating system. so people can give 5 star for a good project or low stars for bad projects, which are not working or has alot errors.
